doc/src/site/sphinx/tutorials/sw_google_cloud_dataproc.rst

```diff
@@ -0,0 +1,184 @@
+
+4. Copy and save the following customization script. This will be required in a later step.
+
+ ::
+
+	#!/bin/bash
+
+	apt-get update
+	apt-get install -y python-dev python-pip jq
+
+	cd /usr/lib/
+	wget http://h2o-release.s3.amazonaws.com/sparkling-water/rel-2.3/8/sparkling-water-2.3.8.zip
+	unzip sparkling-water-2.3.8.zip
+
+	pip install pip==9.0.3
+	pip install requests==2.18.4
+	pip install tabulate
+	pip install future
+	pip install colorama
+	pip install scikit-learn==0.19.1
+	pip install https://h2o-release.s3.amazonaws.com/h2o/rel-wright/4/Python/h2o-3.20.0.4-py2.py3-none-any.whl
+	pip install --upgrade google-cloud-bigquery
+	pip install --upgrade google-cloud-storage
+
+	pip install h2o_pysparkling_2.3
+
+
+5. Create a Dataproc Custom Image as defined `here <https://cloud.google.com/dataproc/docs/guides/dataproc-images>`__. Note that this references the Daisy path, the custom image name, and the customization script from previous steps. The code will be similar to the following:
+
+ ::
+
+   python generate_custom_image.py \
+    --image-name <custom_image_name> \
+    --dataproc-version <Cloud Dataproc version (example: "1.2.22")> \
+    --customization-script <local path to your custom script> \
+    --daisy-path <local path to the downloaded daisy binary> \
+    --zone <Compute Engine zone to use for the location of the temporary VM> \
+    --gcs-bucket <URI (gs://bucket-name) of a Cloud Storage bucket in your project> \
+    [--no-smoke-test: <"true" or "false" (default: "false")>
+
+6. Copy and save the following initialization script into a GCS bucket that you have access to. This will be required when you create your cluster.
+
+ ::
+
+	#!/bin/bash
+
+	METADATA_ROOT='http://metadata/computeMetadata/v1/instance/attributes'
+
+	CLUSTER_NAME=$(curl -f -s -H Metadata-Flavor:Google \
+	${METADATA_ROOT}/dataproc-cluster-name)
+
+	cat << EOF > /usr/local/bin/await_cluster_and_run_command.sh
+	#!/bin/bash
+	# Helper to get current cluster state.
+	function get_cluster_state() {
+		echo \$(gcloud dataproc clusters describe ${CLUSTER_NAME} | \
+	  	grep -A 1 "^status:" | grep "state:" | cut -d ':' -f 2)
+	}
+	# Helper which waits for RUNNING state before issuing the command.
+	function await_and_submit() {
+		local cluster_state=\$(get_cluster_state)
+		echo "Current cluster state: \${cluster_state}"
+		while [[ "\${cluster_state}" != 'RUNNING' ]]; do
+		echo "Sleeping to await cluster health..."
+		sleep 5
+		local cluster_state=\$(get_cluster_state)
+		if [[ "\${cluster_state}" == 'ERROR' ]]; then
+		  echo "Giving up due to cluster state '\${cluster_state}'"
+		  exit 1
+		fi
+		done
+
+		echo "Changing Spark Configurations"
+		sudo sed -i 's/spark.dynamicAllocation.enabled true/# spark.dynamicAllocation.enabled true/g' /usr/lib/spark/conf/spark-defaults.conf
```

Comment:
`spark.dynamicAllocation.enabled true` needs to be changed to `spark.dynamicAllocation.enabled false`

Reply:
Didn't need to be changed from `true` to `false`. sed is replacing the default with `#<commented line>`. I felt it would be better if someone were to dive into the settings to see all the defaults commented out, rather than completely changed/overwritten.